src/main/java/net/minecraftforge/registries/RegistryManager.java

```diff
@@ -133,9 +141,25 @@ public String getName()
             this.synced.add(name);
         for (ResourceLocation legacyName : builder.getLegacyNames())
             addLegacyName(legacyName, name);
+        if (registerToRoot && this == RegistryManager.ACTIVE && builder.getHasWrapper() && !Registry.REGISTRY.containsKey(reg.getRegistryName()) && Registry.REGISTRY instanceof WritableRegistry<?> rootRegistry)
+            registerRegistry(rootRegistry, reg);
         return getRegistry(name);
     }
 
+    @SuppressWarnings("unchecked")
+    private <T> void registerRegistry(WritableRegistry<T> rootRegistry, ForgeRegistry<?> forgeReg) {
+        Registry<?> wrapper = forgeReg.getWrapper();
+        if (wrapper != null)
+            rootRegistry.register((ResourceKey<T>) forgeReg.getRegistryKey(), (T) wrapper, Lifecycle.experimental());
+    }
+
+    public static CompletableFuture<List<Throwable>> preNewRegistryEvent(final Executor executor, final IModStateTransition.EventGenerator<? extends RegistryEvent.NewRegistry> eventGenerator) {
```

Comment:
Formatting: braces on newlines.
```suggestion
    public static CompletableFuture<List<Throwable>> preNewRegistryEvent(final Executor executor, final IModStateTransition.EventGenerator<? extends RegistryEvent.NewRegistry> eventGenerator)
    {
```